tests/wallet/did_wallet/test_did_rpc.py

```diff
@@ -1,123 +1,57 @@
-        rpc_server_cleanup = await start_rpc_server(
-            api_one,
-            self_hostname,
-            daemon_port,
-            test_rpc_port,
-            lambda x: None,
-            bt.root_path,
-            config,
-            connect_to_daemon=False,
+        funds = sum(
+            [
+                calculate_pool_reward(uint32(i)) + calculate_base_farmer_reward(uint32(i))
+                for i in range(1, num_blocks - 1)
+            ]
         )
 
-        async def got_initial_money():
-            balances = await client.get_wallet_balance("1")
-            return balances["confirmed_wallet_balance"] > 0
-
-        await time_out_assert(timeout=MAX_WAIT_SECS, function=got_initial_money)
-
-        val = await client.create_new_did_wallet(201)
-
-        assert isinstance(val, dict)
-        if "success" in val:
-            assert val["success"]
-        assert val["type"] == WalletType.DISTRIBUTED_ID.value
-        assert val["wallet_id"] > 1
-        assert len(val["my_did"]) == 64
-        assert bytes.fromhex(val["my_did"])
-
-        main_wallet_2 = wallet_node_2.wallet_state_manager.main_wallet
-        ph2 = await main_wallet_2.get_new_puzzlehash()
-        for i in range(0, num_blocks + 1):
-            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph2))
-
-        recovery_list = [bytes.fromhex(val["my_did"])]
-        async with wallet_node_2.wallet_state_manager.lock:
-            did_wallet_2: DIDWallet = await DIDWallet.create_new_did_wallet(
-                wallet_node_2.wallet_state_manager, main_wallet_2, uint64(101), recovery_list
-            )
-        for i in range(0, num_blocks):
-            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(32 * b"\0"))
-
-        filename = "test.backup"
-        did_wallet_2.create_backup(filename)
-
-        val = await client.create_new_did_wallet_from_recovery(filename)
-        if "success" in val:
-            assert val["success"]
-        assert val["type"] == WalletType.DISTRIBUTED_ID.value
-        assert val["wallet_id"] > 1
-        did_wallet_id_3 = val["wallet_id"]
-        assert len(val["my_did"]) == 64
-        assert bytes.fromhex(val["my_did"]) == did_wallet_2.did_info.origin_coin.name()
-        assert bytes.fromhex(val["coin_name"])
-        assert bytes.fromhex(val["newpuzhash"])
-        assert bytes.fromhex(val["pubkey"])
-
-        filename = "test.attest"
-        val = await client.did_create_attest(
-            did_wallet_2.wallet_id, val["coin_name"], val["pubkey"], val["newpuzhash"], filename
+        await time_out_assert(10, wallet_0.get_unconfirmed_balance, funds)
+        await time_out_assert(10, wallet_0.get_confirmed_balance, funds)
+        api_0 = WalletRpcApi(wallet_node_0)
+        val = await api_0.create_new_wallet(
+            {"wallet_type": "did_wallet", "did_type": "new", "backup_dids": [], "amount": 1}
         )
+        await asyncio.sleep(2)
```

Comment:
flaky or slow, we should wait for it to enter the mempool or get confirmed, we should not do `asyncio.sleep(x)` in the vast majority of cases

Reply:
FWIW, I'm working on some helpers over in https://github.com/Chia-Network/chia-blockchain/pull/11535.